src/main/java/org/myrobotlab/opencv/OpenCVFilterTextDetector.java

```diff
@@ -0,0 +1,364 @@
+        // Center point of the rect
+        float  centerX = (float) (0.5*(p1_0+p3_0));
+        float  centerY =  (float) (0.5*(p1_1+p3_1));
+        Point2f center = new Point2f(centerX, centerY);
+        // The dimensions of the rect
+        Size2f size = new Size2f((float)w,(float)h);
+        // Create the rotated rect. 
+        // TODO: It would be nice if this was scaled back up to the 
+        // original resolution here.  (perhaps, that means scaling x0/1/2/3_data.. 
+        // and what does that do to the angle if we scale with a non square aspect ratio? ... icky.
+        RotatedRect rec = new RotatedRect(center, size, (float) (-1*angle * 180.0 / Math.PI));
+        boxes.add(rec);
+        confidences.add(score);
+      }
+    }
+    // Apply non-maximum suppression to filter down boxes that mostly overlap
+    ArrayList<DetectedText> maxRects = applyNMSBoxes(threshold, boxes, confidences, nmsThreshold);
+    // This is the filtered list of rects that matched our threshold.
+    classifications = orderRects(maxRects, frame.cols());
+    return maxRects;
+  }
+
+  private ArrayList<DetectedText> orderRects(ArrayList<DetectedText> maxRects, int width) {
+    Comparator<DetectedText> rectComparator = new Comparator<DetectedText>() {         
+      @Override         
+      public int compare(DetectedText rect1, DetectedText rect2) {
+        // left to right.. top to bottom. 
+        // TODO: this 100 is a vertical sort of resolution ... it should be more dynamic
+        // and it should probably be configured somehow..  in reality this algorithm needs to be replaced/fixed
+        int index1 = rect1.box.boundingRect().x() + (rect1.box.boundingRect().y()*width/100); 
+        int index2 = rect2.box.boundingRect().x() + (rect2.box.boundingRect().y()*width/100);
+        return (index2 > index1 ? -1 : (index2 == index1 ? 0 : 1));           
+      }     
+    };    
+    maxRects.sort(rectComparator);
+    return maxRects;
+  }
+
+  private static ArrayList<DetectedText> applyNMSBoxes(float threshold, ArrayList<RotatedRect> boxes, ArrayList<Float> confidences, float nmsThreshold) {    
+    RectVector boxesRV = new RectVector();
+    for (RotatedRect rr : boxes) {
+      boxesRV.push_back(rr.boundingRect());
+    }
+    FloatPointer confidencesFV = arrayListToFloatPointer(confidences);
+    IntPointer indicesIp = new IntPointer();
+    NMSBoxes(boxesRV, confidencesFV, (float)threshold, nmsThreshold, indicesIp);
+    ArrayList<DetectedText> goodOnes = new ArrayList<DetectedText>();
+    for (int m=0;m<indicesIp.limit();m++) {
+      int i = indicesIp.get(m);
+      RotatedRect box = boxes.get(i); 
+      confidencesFV.position(i);
+      // we don't have text yet, that will be filled in later by the ocr step.
+      DetectedText dt = new DetectedText(box, confidencesFV.get(), null);
+      goodOnes.add(dt);
+    }
+    return goodOnes;
+  }
+
+  // utilty helper function to put an array of floats into a javacpp float pointer
```

Comment:
perhaps generalized type conversion methods should be promoted to the org.myrobotlab.opencv as helper methods other filters can use ?

Reply:
move them to the current image.Util class.. we can move them around somewhere later if we like.